src/doctor.ts

```diff
@@ -24,8 +31,10 @@ export class DoctorProvider implements Disposable {
    * We should take focus into account when doctor caching mechanism will be implemented on the server.
    */
   isOpened = false;
-
-  constructor(private client: LanguageClient) {}
+  private extensionContext: ExtensionContext | undefined;
+  constructor(private client: LanguageClient, context: ExtensionContext) {
+    this.extensionContext = context;
+  }
```

Comment:
Why union type with undefined? Also, if you add modifier to the constructor parameter, it'll be converted to class field automatically. 
so
```typescript
constructor(private client: LanguageClient) {}
```
is more or less equal to
```typescript
  client: LanguageClient
  constructor(client: LanguageClient) {
    this.client = client;
  }
```

```suggestion
  doctorIconPath: Uri;

  constructor(
    private client: LanguageClient,
    extensionContext: ExtensionContext
  ) {
    this.doctorIconPath = Uri.file(
      path.join(extensionContext.extensionPath, "icons", "doctor.svg")
    );
  }
```

Reply:
> Why union type with undefined? Also, if you add modifier to the constructor parameter, it'll be converted to class field automatically.

it's just because my typescript knowledge is super limited :sweat_smile:

Reply:
Fixed, thanks!